### ISSUE TITLE
Handling Fixed Headers with Offset

### ISSUE THREAD

**Issue by jonmcgill** [bvaughn/react-resizable-panels#152]
I've got a vertical `PanelGroup` where each `Panel` has a header right above it and a resizer right below it. I want the headers to remain at a fixed height but also control whether or not the panel below it shows or hides. Everything works well but when I resize any Panel it's (understandably) not taking into consideration the height of the headers and the resize percentages get out of sync with the user's cursor position.

https://github.com/bvaughn/react-resizable-panels/assets/13082333/cc4bf2fa-6fee-490c-89fe-c041efd355e2

Implementation looks something like this:
```tsx
<PanelGroup>
  <div>Header1</div>
  {open1 && (
    <Panel>
    <Resizer>
  )}
  <div>Header2</div>
  {open2 && (
    <Panel>
    <Resizer>
  )}
</PanelGroup>
```

First, am I just getting the implementation wrong and this is something I can accomplish with the current API?
Second, if not, would it be possible to add some option or config that would tell the internals to offset its calculation by a certain fixed amount?

**Comment by bvaughn** [comment]
Can you link me to something I can run and tinker with (Code Sandbox, Stackblitz, whatever)

**Comment by jonmcgill** [comment]
Sure thing, thanks for taking a look: https://codesandbox.io/s/bold-montalcini-n72wcn?file=/src/App.tsx

**Comment by bvaughn** [comment]
Cool. Thanks! tI think [his is how I would do it](https://codesandbox.io/s/beautiful-minsky-7ov4k2).

```ts
function PanelWithHeader({
  children,
  header,
  order
}: PropsWithChildren<{ header: string; order: number }>) {
  const [showPanel, setShowPanel] = useState(true);

  return showPanel ? (
    <>
      <Panel order={order}>
        <div className="header" onClick={() => setShowPanel(false)}>
          {header}
        </div>
        <div className="panel">{children}</div>
      </Panel>
      <PanelResizeHandle>
        <div className="resize" />
      </PanelResizeHandle>
    </>
  ) : (
    <div className="header" onClick={() => setShowPanel(true)}>
      <span>{header}</span>
    </div>
  );
}

export default function App() {
  return (
    <div className="wrap">
      <PanelGroup direction="vertical">
        <PanelWithHeader children="Panel 1" header="Panel 1" order={1} />
        <PanelWithHeader children="Panel 2" header="Panel 2" order={2} />
        <PanelWithHeader children="Panel 3" header="Panel 3" order={3} />
      </PanelGroup>
    </div>
  );
}
```

**Comment by jonmcgill** [comment]
Ok, so this still renders a non-Panel child with a fixed height within the PanelGroup if that panel happens to be closed. I suppose the fundamental question I'm asking is how to (or can we) have non-Panel children with specific heights inside a PanelGroup in a way that doesn't mess with the panel height calculation on resize.

This illustrates the point, I think:
```tsx
export default function App() {
return (
    <div className="wrap">
      <PanelGroup direction="vertical">
        {/* this 200px height is not taken into consideration when resizing.
            it would be helpful to instruct the resizing calculation to subtract 200
            from the total height when doing the resizing into percentages */}
        <div style={{ height: 200, background: "pink" }} />
        <Panel order={1} className="panel">
          <span>Panel 1 content</span>
        </Panel>
        <PanelResizeHandle>
          <div className="resize" />
        </PanelResizeHandle>
        <Panel order={2} className="panel">
          <span>Panel 2 content</span>
        </Panel>
        <PanelResizeHandle>
          <div className="resize" />
        </PanelResizeHandle>
        <Panel order={3} className="panel">
          <span>Panel 3 content</span>
        </Panel>
        <PanelResizeHandle>
          <div className="resize" />
        </PanelResizeHandle>
      </PanelGroup>
    </div>
  );
}
```

**Comment by bvaughn** [comment]
> Ok, so this still renders a non-Panel child with a fixed height within the PanelGroup if that panel happens to be closed.

You mean if the panel is collapsible? No, it should be hidden in that case.

I think I'm going to need you to share a repro of what you're talking about. Textual descriptions aren't enough.

**Comment by jonmcgill** [comment]
The repo can't be public, sorry. I think I may have confused things with my previous example. I'll try and use your initial response to explain what I'm after.

In your `PanelWithHeader` component, if Panel 2 was closed and the others were open, you'd still have something like this rendered:

```tsx
<PanelGroup>
  <Panel>
    <div className="header">Panel 1</div>
  </Panel>
  <PanelResizeHandle />
  <div className="header">Panel 2</div>
  <Panel>
    <div className="header">Panel 3</div>
  </Panel>
  <PanelResizeHandle />
</PanelGroup>
```

Let's say the `PanelGroup` element had a height of `500px` and each `.header` element had a height of `50px`. When resizing Panel 1 the height is going to be based on `500px` total but with that Panel 2 header still a child, the actual height to base resizing on is `450px`.

It's fine, I think what I'm looking to do is not necessarily supported and I may have to just find a work-around.

**Comment by bvaughn** [comment]
That's okay. I don't need to see your actual application code, but it's hard for me to provide helpful feedback without a _repro_ of the constraints you're working with. Unfortunately, without that, I don't understand why this would be a problem:

> When resizing Panel 1 the height is going to be based on 500px total but with that Panel 2 header still a child, the actual height to base resizing on is 450px.

The header has to take up some space if it's visible, right?

**Comment by jonmcgill** [comment]
Thanks again for your dialogue here. I did end up figuring out how to achieve what I needed. I added logic to my implementation to check which sections were closed and I added those sections' headers to the closest open panel below it. This means that the last section in the group needed to check for all the sections above it to see if if needed to render a closed header, but it works perfectly.

Thanks again for this great library, it is 🤌
